File: LeetCode/CountPrimes/countPrimes.ts
function countPrimes(n: number): number {
  let primes = Array(n).fill(1);
  let primeCount = 0;
  let limit = Math.sqrt(n);

  if (n <= 2) {
    return 0;
  }

  primes[0] = 0;
  primes[1] = 0;
  for (let i = 2; i <= limit; i++) {
    if (primes[i] === 1) {
      for (let j = i * i; j < n; j += i) {
        primes[j] = 0;
      }
    }
  }

  primeCount = primes.reduce((acc, ele) => acc + ele, 0);

  return primeCount;
}

File: LeetCode/CountPrimes/CountPrimes.cs
public class Solution {
    public int CountPrimes(int n) {
        var Primes = new int[n];
        var PrimeCount = 0;
        Array.Fill(Primes,1);
        int limit = (int)Math.Sqrt(n);

        if(n<=2){
            return 0;
        }

        for(int i = 2; i<=limit; i++){
            if(Primes[i] == 1){
                for(int j = i*i; j<n; j+=i){
                    Primes[j] = 0;
                }
            }
        }

        foreach(int ele in Primes){
            if(ele == 1){
                PrimeCount++;
            }
        }

        return PrimeCount-2;
    }
}